packages/fetchai/connections/ledger/contract_dispatcher.py

```diff
@@ -355,12 +355,22 @@ def _validate_and_call_callable(
         :param contract: the contract instance.
         :return: the data generated by the method.
         """
+        kwargs = message.kwargs.body
         try:
             method_to_call = getattr(contract, message.callable)
         except AttributeError:
-            raise AEAException(
-                f"Cannot find {message.callable} in contract {type(contract)}"
-            )
+            # Check for the method in the ABI
+            try:
+                contract_instance = contract.get_instance(api, message.contract_address)
+                contract_instance.get_function_by_name(message.callable)
+            except ValueError:
+                raise AEAException(
+                    f"Cannot find {message.callable} in contract {type(contract)}"
+                )
```

Comment:
lets log that we call the abi directly instead of the contract. Also I don't like nested try / excepts, can do this instead:

```
      try:
          method_to_call = getattr(contract, message.callable)
      except AttributeError:
          logging.info(f"Cannot find {message.callable} in contract {type(contract)}, checking for method in the ABI")
      try:
          contract_instance = contract.get_instance(api, message.contract_address)
          contract_instance.get_function_by_name(message.callable)
      except ValueError:
          raise AEAException(
              f"Cannot find {message.callable} in contract {type(contract)}"
          )
```

Reply:
The second try-except block must be conditional as if we don't get an exception we don't want to run that. So either I nest or I set an if block where I check `if method_to_call:`